cmake/partition_manager.cmake

```diff
@@ -47,19 +59,62 @@ if(PM_IMAGES OR (EXISTS ${static_configuration_file}))
   # Add subsys defined pm.yml to the input_files
   list(APPEND input_files ${PM_SUBSYS_PREPROCESSED})
 
+  math(EXPR ram_size "${CONFIG_SRAM_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
+
+  if (DEFINED CONFIG_SOC_NRF9160)
+    # See nRF9160 Product Specification, chapter "UICR"
+    set(otp_start_addr "0xff8108")
+    set(otp_size 756) # 189 * 4
+  elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
+    # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
+    set(otp_start_addr "0xff8100")
+    set(otp_size 764)  # 191 * 4
+  endif()
+
+  math(EXPR flash_size "${CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
+
+  if (CONFIG_SOC_NRF9160 OR CONFIG_SOC_NRF5340_CPUAPP)
+    add_region(
```

Comment:
Not sure how it works, but we are writing to OTP, so do we need NRF_FLASH_DRV_NAME for that as well?

Reply:
No, AFAIK we don't use the nrf_soc_flash.c driver when operating on the UICR.